hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -110,8 +111,9 @@
 @SuppressWarnings("checkstyle:classfanoutcomplexity")
 public final class ProxyManager {
 
-    private static final String PROVIDER_ID = "com.hazelcast.client.spi.ClientProxyDescriptorProvider";
-    private static final Class[] CONSTRUCTOR_ARGUMENT_TYPES = new Class[]{String.class, String.class};
+    private static final String PROVIDER_ID = ClientProxyDescriptorProvider.class.getCanonicalName();
+    private static final Class[] LEGACY_CONSTRUCTOR_ARGUMENT_TYPES = new Class[]{String.class, String.class};
```

Comment:
I kept the old signature for legacy implementations which are read via the `ClientProxyDescriptorProvider`.